tests/test_views.py

```diff
@@ -0,0 +1,1050 @@
+        )
+
+        expected_context = {
+            'scope_form': views.ScopeForm(),
+            'form_errors': None,
+            'summary_list': None,
+            'json_data': None
+        }
+        response = client.get(reverse('url_listing', args=[project.project_slug, entity]))
+
+        for key in expected_context:
+            assert str(response.context[key]) == str(expected_context[key])
+
+
+class TestUrlSurt():
+
+    def test_status_ok(self, rf):
+        project = factories.ProjectFactory()
+        request = rf.get('/')
+        response = views.url_surt(request, project.project_slug, 'a_surt')
+
+        assert response.status_code == 200
+
+    def test_template_used(self, client):
+        project = factories.ProjectFactory()
+        response = client.get(reverse('url_surt', args=[project.project_slug, 'a surt']))
+        assert response.templates[0].name == 'nomination/url_surt.html'
+
+    @pytest.mark.parametrize('surt_root, surt, url_list_len, letter', [
+        ('http://(com,e', 'http://(com,example,www)', 1, 'e'),
+        ('http://(com,example,www)', 'http://(com,example,www)', 1, False),
+        ('http://(com,a', 'http://(com,example,www)', 0, 'a'),
+        ('http://(com,apples,www)', 'http://(com,example,www)', 0, False)
+    ])
+    def test_context(self, client, surt_root, surt, url_list_len, letter):
+        project = factories.ProjectFactory()
+        url = factories.URLFactory(
+            url_project=project,
+            attribute='surt',
+            value=surt
+        )
+        response = client.get(reverse('url_surt', args=[project.project_slug, surt_root]))
+
+        assert response.context['surt'] == surt_root
+        assert response.context['project'] == project
+        assert len(response.context['url_list']) == url_list_len
+        if url_list_len == 1:
+            assert response.context['url_list'][0] == url
+        assert response.context['letter'] == letter
+        assert response.context['browse_domain'] == 'com'
+        assert len(response.context['browse_dict']) == 1
+        for each in response.context['browse_dict']['com']:
+            if each[1] is not None:
+                assert each == ('E', 'http://(com,e')
+
+
+class TestUrlAdd():
+
+    @pytest.mark.parametrize('request_type, data_dict', [
+        ('get', {}),
+        ('post', {
+            'nominator_name': 'Eddie',
+            'nominator_institution': 'UNT',
+            'nominator_email': 'someone@somewhere.com'
+        })
+    ])
+    def test_status_ok(self, client, request_type, data_dict):
+        project = factories.ProjectFactory()
+        response = getattr(client, request_type)(
+            reverse('url_add', args=[project.project_slug]), data_dict)
+        assert response.status_code == 200
+
+    def test_template_used(self, client):
+        project = factories.ProjectFactory(registration_required=False)
+        response = client.post(
```

Comment:
Is there a reason to issue a POST request vs a GET request? A GET seems like it would be easier.